internal/services/rate.go

```diff
@@ -68,29 +76,37 @@ func (rs *CachingRateService) StartBackgroundTask() {
 	}()
 }
 
-func (rs *CachingRateService) GetRate() (*Rate, error) {
+func (rs *CachingRateService) GetRate() (float32, error) {
 	rs.mutex.RLock()
 	defer rs.mutex.RUnlock()
 
 	if rs.fetchError != nil {
-		return nil, rs.fetchError
+		return 0, rs.fetchError
 	}
 	return rs.currentRate, nil
 }
 
-func (rs *CachingRateService) fetchExchangeRate() (*Rate, error) {
-	resp, err := http.Get(RatesAPIBaseURL + "/usd.json")
+type HTTPExchangeRateClient struct {
```

Comment:
Good improvement 👍

Reply:
Is it expected this client will be instantiated without constructor (`NewHTTPExchangeRateClient`)?
If it's not, I don't see a reason to make this struct available outside of the package.

P.S. the same comment to other structs in the repo)

Reply:
This topic is discussed here https://github.com/golang/lint/issues/210. And I agree, for better encapsulation I should make structs unexported, but revive linter gives me warning like this `unexported-return: exported func NewRateService returns unexported type *services.cachingRateService, which can be annoying to use`. So I will just ignore these warnings for now.